src/backend/catalog/namespace.c

```diff
@@ -3629,15 +3629,10 @@ GetOverrideSearchPath(MemoryContext context)
 	result = (OverrideSearchPath *) palloc0(sizeof(OverrideSearchPath));
 	schemas = list_copy(activeSearchPath);
 	compatibleNS = get_namespace_oid("oracle", true);
-	while (schemas && linitial_oid(schemas) != activeCreationNamespace)
+	while (schemas && (linitial_oid(schemas) != activeCreationNamespace && linitial_oid(schemas) != compatibleNS))
 	{
 		if (linitial_oid(schemas) == myTempNamespace)
 			result->addTemp = true;
-		else if(OidIsValid(compatibleNS) && linitial_oid(schemas) == compatibleNS)
-		{
-			schemas = list_delete_first(schemas);
-			continue;
-		}
```

Comment:
I think this is breaking the existing functionality of this function. GetOverrideSearchPath() function removes the implicitly added namespaces. But with this check, we are preventing it from doing that. 

GetOverrideSearchPath() should not have these checks here, this logic should go to the PushOverrideSearchPath() instead. like
```
if (compatible_db == COMPATIBLE_ORA)
 {
 		compatibleNS = get_namespace_oid("oracle", true);
 		if (OidIsValid(compatibleNS) &&
 			!list_member_oid(oidlist, compatibleNS))
 			oidlist = lcons_oid(compatibleNS, oidlist);
}
```

Reply:
I think GetOverrideSearchPath

> I think this is breaking the existing functionality of this function. GetOverrideSearchPath() function removes the implicitly added namespaces. But with this check, we are preventing it from doing that.
> 
> GetOverrideSearchPath() should not have these checks here, this logic should go to the PushOverrideSearchPath() instead. like
> 
> ```
> if (compatible_db == COMPATIBLE_ORA)
>  {
>  		compatibleNS = get_namespace_oid("oracle", true);
>  		if (OidIsValid(compatibleNS) &&
>  			!list_member_oid(oidlist, compatibleNS))
>  			oidlist = lcons_oid(compatibleNS, oidlist);
> }
> ```

the GetOverrideSearchPath() function only prevent to delete the oralce schema,and it will not affect others.

Reply:
is it really needed to add 'addora' variable? if "compatible_db == COMPATIBLE_ORA" is true that means we have already added oracle schema implicitly before calling GetOverrideSearchPath().

So maybe we should just add it again in PushOverrideSearchPath() when "compatible_db == COMPATIBLE_ORA" is true.